test/KnoBlockIO.ts

```diff
@@ -1,159 +1,265 @@
-      it('Should set the right owner', async function () {
-        await KnoBlockIOInstance.deployed();
-        const owner = await KnoBlockIOInstance.MockReturnOwner();
-        expect(deployer.address).to.equal(owner);
+      before(async function () {
+        [deployer, bob, alice] = await ethers.getSigners();
       });
-    });
 
-    describe('Create KnoBlock', function () {
-      //Is there not a way to initiate the function once for all these checks
-      it('Should increment count variable by 1', async function () {
-        await KnoBlockIOInstance.deployed();
-        await KnoBlockIOInstance.createKnoBlock(1001, 1);
-        expect(await KnoBlockIOInstance.MockReturnCount()).to.equal(1);
+      beforeEach(async function () {
+        instance = await deploy();
       });
 
-      it('Should add a new KnoBlock to the KnoBlock Mapping', async function () {
-        await KnoBlockIOInstance.deployed();
-        await KnoBlockIOInstance.createKnoBlock(1001, 1);
-        expect(await KnoBlockIOInstance.JungleIsMassiv).to.equal(1);
+      describe('Deployment', function () {
+        it('sets the right owner', async function () {
+          const owner = await instance.connect(deployer).owner();
+          expect(deployer.address).to.equal(owner);
+        });
       });
 
-      it('new KnoBlock should have msg.sender as creator', async function () {
-        await KnoBlockIOInstance.deployed();
-        await KnoBlockIOInstance.createKnoBlock(1001, 1);
-        expect(await KnoBlockIOInstance.MockReturnKnoBlockCreator(0)).to.equal(
-          deployer.address,
-        );
+      describe('#create()', () => {
+        it('increments the count variable by 1', async function () {
+          await instance.connect(deployer).create(1001, one);
+          expect(await instance.connect(deployer).count()).to.equal(one);
+        });
+
+        it('sets creator as msg.sender', async function () {
+          await instance.connect(deployer).create(1001, one);
+          expect(await instance.connect(deployer).creator(zero)).to.equal(
+            deployer.address,
+          );
+        });
+
+        it('sets unlockAmount correctly', async function () {
+          await instance.connect(deployer).create(1001, one);
+          expect(await instance.connect(deployer).unlockAmount(zero)).to.equal(
+            BigNumber.from('1001'),
+          );
+        });
+
+        it('sets knoType correctly', async function () {
+          await instance.connect(deployer).create(1001, one);
+          expect(await instance.connect(deployer).knoType(zero)).to.equal(one);
+        });
+        it('emits new KnoBlock Event', async function () {
+          await expect(instance.connect(deployer).create(1001, one))
+            .to.emit(instance, 'KnoBlockCreated')
+            .withArgs(zero);
+        });
       });
+      describe('#deposit()', () => {
```

Comment:
Include all the function signature:
```suggestion
      describe('#deposit(uint256,uint256)', () => {
```